### PR TITLE
add rails example

### LINKED ISSUE #146: Best way to integrate with a rails project?
What is the best way to integrate JS error reporting into my rails project?  Some options:

1. Copy this file into my local project: https://raw.githubusercontent.com/airbrake/airbrake-js/master/dist/client.js
2. Include from CDN, eg https://ssljscdn.airbrake.io/airbrake.min.js

I would think # 2, as that should always give me the latest greatest version of your code?

BTW, I got # 2 by modifying a URL found in [this article](https://airbrake.io/blog/notifier/airbrake-ruby-gem-3-1-5-airbrake-js-0-3-beta).

Thanks!

### PR DESCRIPTION
**Pull request by shime**
closes #146

### CODE REVIEW

**Comment by vmihailenco** [comment]
This requires some README explaining how to start your example, e.g. https://github.com/airbrake/airbrake-js/blob/master/examples/requirejs/README.md. Also notifier copy should not be included in example, e.g. file examples/rails/app/assets/javascripts/node_modules/airbrake-js/dist/client.js should be removed.

Also I expect from such example to show some Rails specific things, e.g.:
- loading airbrake project id, API key and environment from Rails config
- simultaneously usage of airbrake-js and https://github.com/airbrake/airbrake
- probably using rails-assets-airbrake-js-client as explained at https://github.com/airbrake/airbrake-js/issues/146#issuecomment-148364695

Probably @kyrylo has more ideas what settings airbrake-js should take from Rails notifier.

**Comment by shime** [comment]
I've written the README and removed `node_modules` in favor of Rails Assets. I've also set it up to load configuration from `ENV`. 

I think using `airbrake` for backend here would only confuse people so I've left it out.

**Comment by vmihailenco** [comment]
So far I don't see the purpose of this example. The only Rails specific thing here is using Sprockets, which actually can be used without Rails. That said the only interesting file is https://github.com/airbrake/airbrake-js/blob/feature/add-rails-example/examples/rails/app/assets/javascripts/application.js.erb. The rest is default Rails app. I think you need to add more Rails specific things like interoperation with airbrake-ruby/airbrake-rails to make this useful.

**Comment by zefer** [comment]
I agree with @vmihailenco, all these default Rails app files serve only to confuse people, and hide the the useful example. One solution may be to _only_ provide the useful example files, e.g. `examples/rails/app/assets/javascripts/application.js.erb` and `examples/rails/config/initializers/airbrake.rb` or whatever they may be?

**Comment by shime** [comment]
@zefer these default Rails app files are here in order for this example to be executable, if I remove them people won't be able to run this example.

**Comment by zefer** [comment]
The original question here https://github.com/airbrake/airbrake-js/issues/146#issue-89381911 was "What is the best way to integrate JS error reporting into my rails project?". I think that's a very valid question, we should strive to make this more clear.

> @zefer these default Rails app files are here in order for this example to be executable, if I remove them people won't be able to run this example.

Does being able to run the example help understand the best way to integrate Airbrake with Rails? Personally, I don't think it does, because it hides all the specifics amongst all the Rails default files.

**Comment by shime** [comment]
@zefer, @vmihailenco I've now removed all the Rails specific files and left only the readme with the description about what needs to be added inside a Rails application.

File: examples/rails/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+end
+```
+
+And then place the following code into your `application.js.erb`:
+
+```javascript
+//= require airbrake-js-client
+
+var airbrake = new airbrakeJs.Client({projectId: 1234, projectKey: 'changeme'});
+airbrake.addFilter(function(notice){
```
Comment (vmihailenco) [inline examples/rails/README.md]: space before `{`

File: examples/rails/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  notice.context.environment = "<%= Rails.env %>";
+  return notice;
+});
+
+try {
+  throw new Error('hello from airbrake-js');
+} catch (err) {
+  promise = airbrake.notify(err);
```
Comment (vmihailenco) [inline examples/rails/README.md]: var promise

**Comment by vmihailenco** [comment]
LGTM

**Comment by zefer** [comment]
:+1: LGTM

**Pull request merged**